From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 5 Jun 2017 13:30:27 +0200
Subject: [PATCH] {BUG} atomisp2: do not redefine CEIL_DIV
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 7.1.1 reports (with CEIL_DIV updated in math_support.h):

    drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c:
    In function ‘__configure_capture_pp_input’:
    drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c:2509:0:
    error: "CEIL_DIV" redefined [-Werror]
     #define CEIL_DIV(a, b)       ((b) ? ((a) + (b) - 1) / (b) : 0)

    In file included from
    drivers/staging/media/atomisp/pci/atomisp2/css2400/sh_css_internal.h:19:0,
    from
    drivers/staging/media/atomisp/pci/atomisp2/css2400/runtime/debug/interface/ia_css_debug.h:28,
    from
    drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c:43:
    drivers/staging/media/atomisp/pci/atomisp2/css2400/hive_isp_css_include/math_support.h:67:0:
    note: this is the location of the previous definition
     #define CEIL_DIV(a, b)       ((b) != 0 ? ((a) + (b) - 1) / (b) : 0)
---
 drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c b/drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c
index 157554de4ea6..4c353ff5e6ec 100644
--- a/drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c
+++ b/drivers/staging/media/atomisp/pci/atomisp2/atomisp_compat_css20.c
@@ -2506,7 +2506,6 @@ static void __configure_capture_pp_input(struct atomisp_sub_device *asd,
 	struct ia_css_pipe_extra_config *pipe_extra_configs =
 		&stream_env->pipe_extra_configs[pipe_id];
 	unsigned int hor_ds_factor = 0, ver_ds_factor = 0;
-#define CEIL_DIV(a, b)       ((b) ? ((a) + (b) - 1) / (b) : 0)
 
 	if (width == 0 && height == 0)
 		return;
-- 
